engine/hooks/branch_deletion.go

```diff
@@ -15,3 +15,21 @@ func (s *Service) doBranchDeletionTaskExecution(t *sdk.TaskExecution) (*sdk.Work
 
 	return nil, sdk.WrapError(err, "cannot mark to delete workflow runs")
 }
+
+func (s *Service) stopBranchDeletionTask(branch string) error {
+	keys, err := s.Dao.FindAllKeysMatchingPattern(branch + "*")
```

Comment:
could you add a check to check if branch is not empty (to avoid drop everything)

Reply:
Good idea ! Haha